### DISCUSSION TITLE
Future Contributions

### DISCUSSION THREAD

**Discussion by CuteSC2** [lamikr/rocm_sdk_builder#159] [Ideas]
Hi, I stumbled upon this project and want to contribute and add additional Features
Thinks I want to do:

Additional GPU support and improvements:
- GFX902/GFX903 (Ryzen 2200g)
- GFX906/907 (Vega Pro VII, Vega VII, Instinct MI50 32GB)
- GFX908 (Instinct MI100)
- GFX90c (Ryzen 5500GT)
- GFX1100 Improvements?

CPU Optimized Builds
- Zen4 (Threadripper 7960X)
- Zen3
- Zen2 (AMD EPYC)
- Zen+

Automated Test scripts to check specific functionality
More Benchmarks

I'll tackle these tasks this Winter.
Any suggestions and input is greatly welcomed.

**Comment by lamikr** [comment]
I agree that Zen4/Zen3 optimization would be really nice to have. Especially I do not have not checked/not have any idea how to enable the avr-512 support. At the moment I do not know whether clang/llvm needs so extra patches for that or build flags or does it get enabled automatically if you make a build for zen4 as a target.

MI100 would be really cool to add.

I have myself access to 2400g and I have had it working long time ago. At that time kernel needed also patches to sysfs methods to get information about available memory, fan speed control etc for all GPUs when used with rocm.
But 2400g particularly needed also some kernel patches, otherwise it failed on memory allocation for gpu kernels.
I can propably check at some point whether I can get the 2400g running and then the 2200g patches would be pretty similar. 2200g and 2400g just starts to be so old iGPUs at the moment that their usability is questionable for anything else than teaching purposes.
